src/libraries/System.Private.CoreLib/src/System/Enum.cs

```diff
@@ -148,8 +148,8 @@ private static string ValueToHexString(object value)
 
         private static string? InternalFlagsFormat(EnumInfo enumInfo, ulong resultValue)
         {
-            string[] names = enumInfo.Names;
-            ulong[] values = enumInfo.Values;
+            string[] names = enumInfo.Names!;
```

Comment:
It would be nice to make Names and Values properties that automatically lazy-initialize the value so that these `!` are unnecessary.

It would also allow unification of `GetEnumInfo` and `GetEnumInfoForActivation` into one method. The split between  `GetEnumInfo` and `GetEnumInfoForActivation` is odd.